### PR TITLE
Fix #25: Support for sidecars and init containers.

### LINKED ISSUE #25: Add possibility to inject self-managed side-cars
**Is your feature request related to a problem? Please describe.**
It would be helpful to add a possibility to inject self-managed sidecar containers (e.g fluentd-ui) into the daemonset. At the moment there is only 1 fixed sidecar container (es-proxy).

**Describe the solution you'd like**
 Add possibility to inject self-managed side-cars

**Describe alternatives you've considered**
Of course it would also be possible to write my own template. But it's way more tricky to get it running properly (As there are more network hops when local-host cannot be used).

**Additional context**

### PR DESCRIPTION
**Pull request by tillig**
Signed-off-by: Travis Illig <tillig@paraesthesia.com>

# Which chart
`fluentd-elasticsearch` (@monotek)

# What this PR does / why we need it
Enables sidecar and init container support in a similar fashion to the [ElasticSearch charts](https://github.com/elastic/helm-charts/blob/09828a9ce85346d34e1bd27e1b8072423163b978/elasticsearch/templates/statefulset.yaml#L369). This is useful in situations where you need sidecars similar to the AWS signing sidecar but which aren't specific to AWS. For example, Istio service mesh can be used to enable TLS for ElasticSearch and manage the certificates - adding the sidecar manually and mounting a shared volume is how you get the certificates in here for use.

# Which issue this PR fixes

- fixes #25

# Special notes for your reviewer

none

# Checklist

- [x] [DCO](https://github.com/kokuwaio/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] All variables are documented in the charts README

### CODE REVIEW

**Review by monotek — CHANGES_REQUESTED** [review]
Imho, this is a bit redundant now, as we already have https://github.com/kokuwaio/helm-charts/blob/main/charts/fluentd-elasticsearch/values.yaml#L18

I guess we could remove the "awsSigningSidecar" now, as it can be configured by the "extraContainers".

As this would be a breaking change, chart major version needs to be updated.

The "awsSigningSidecar" config could be added as commented out example to the values of "extraContainers".

Thoughts?

**Comment by tillig** [comment]
I think removing `awsSigningSidecar` and using that as the example configuration seems like a good way forward. I think `extraContainers` is probably a better general way forward since not only does it remove confusion (I'm not using an AWS signing sidecar, but... that's what I'm configuring?) but it also allows more than one sidecar to be set up.

Let me fix this up.

File: charts/fluentd-elasticsearch/README.md [hunk 1]
```diff
@@ -209,7 +202,7 @@ extraVolumes: |
 
 ### AWS Elasticsearch Domains
 
-AWS Elasticsearch requires requests to upload data to be signed using [AWS Signature V4](https://docs.aws.amazon.com/general/latest/gr/signature-version-4.html). In order to support this, you can add `awsSigningSidecar: {enabled: true}` to your configuration. This results in a sidecar container being deployed that proxies all requests to your Elasticsearch domain and signs them appropriately.
+AWS Elasticsearch requires requests to upload data to be signed using [AWS Signature V4](https://docs.aws.amazon.com/general/latest/gr/signature-version-4.html). In order to support this, you can add a sidecar to the `extraSidecars` configuration. An example is provided in `values.yaml`. This results in a sidecar container being deployed that proxies all requests to your Elasticsearch domain and signs them appropriately.
```
Comment (monotek) [inline charts/fluentd-elasticsearch/README.md:205]:
```suggestion
AWS Elasticsearch requires requests to upload data to be signed using [AWS Signature V4](https://docs.aws.amazon.com/general/latest/gr/signature-version-4.html). In order to support this, you can add a sidecar to the `extraContainers` configuration. An example is provided in `values.yaml`. This results in a sidecar container being deployed that proxies all requests to your Elasticsearch domain and signs them appropriately.
```

**Review by monotek — CHANGES_REQUESTED** [review]

**Comment by monotek** [comment]
Please also fix linting errors...

**Comment by tillig** [comment]
They certainly don't make it easy to see [how to run that chart-testing tool as a Docker image](https://github.com/helm/chart-testing). :( Working on it.

**Comment by tillig** [comment]
OK, got the linting passing. Sorry for the delay.

**Review by monotek — CHANGES_REQUESTED** [review]
Please also extend: https://github.com/kokuwaio/helm-charts/blob/main/charts/fluentd-elasticsearch/README.md#upgrading

**Review by monotek — APPROVED** [review]

**Pull request merged**